pkg/controllers/generic/import_syncer.go

```diff
@@ -244,7 +265,7 @@ func (s *importer) Sync(ctx *synccontext.SyncContext, pObj client.Object, vObj c
 
 	// check if either object is getting deleted
 	if vObj.GetDeletionTimestamp() != nil || pObj.GetDeletionTimestamp() != nil {
-		if pObj.GetDeletionTimestamp() == nil {
+		if pObj.GetDeletionTimestamp() == nil && !s.syncerOptions.IsClusterScopedCRD {
```

Comment:
Could we also add the same condition around the `ctx.PhysicalClient.Delete()` call in the SyncUp() ?
I know that it will not be called because the annotation is not added, but since we don't have any tests, adding a condition could decrease the chance we allow the Deletion to happen after some future code changes.
Actually, now that I mention tests, we should probably have some at least to ensure that Delete / Update does not happen, as the consequences could be quite bad. Wdyt?